File: 2ºCARRERA/tprog/practica2_872815_873338/c++/write.cc
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : write.cc
 */

#include "write.h"

void Write::ejecuta(stack<int32_t> &pila, int& pc) {
    int32_t num = pila.top();
    pila.pop();
    cout << num << endl;
    pc++;
}

void Write::muestra(){
    cout << "write";
}

File: 2ºCARRERA/tprog/practica2_872815_873338/java/Write.java
/*
 *  Asignatura : Tecnología de la Programación
 *  Práctica 2:
 *  Autores : Mateo Lorente, Diego  --- 873338
 *            Solana Melero, Carlos --- 872815
 *  Fichero : Write.java
 */

import java.util.Stack;

class Write implements Instruccion
{
    public int ejecuta(Stack<Integer> pila, int pc) {
        int numero;
        numero = pila.pop();
        System.out.println(numero);
        pc++;
        return pc;
    }
    public void muestra() {
        System.out.print("write");
    }
}